nengo/builder/operator.py

```diff
@@ -388,7 +388,10 @@ def make_step(self, signals, dt, rng):
 
         if inc:
             def step_copy():
-                dst[dst_slice] += src[src_slice]
+                if type(dst_slice) == type(Ellipsis):
```

Comment:
type(dst_slice) == type(Ellipsis) if no slice is provided. So we'd think that dst = dst[dst_slice]
Yet, dst += src[src_slice] sometimes fails because "dst referenced before assignment" because of some weird numpy internal

Reply:
I think `isinstance(dst_slice, Ellipsis)` is usually preferred for type checks (except when derived types should be explicitly excluded which is very rare).

Reply:
Also, the slices should not change during the simulation. That should allow to move the if-branching outside of the `step_copy` function. That might have some performance impact, but no idea whether that's significant.

I'm also curious if there is a performance impact replacing `+=` with `at`.

Reply:
I agree with Jan, the slices won't change during the simulation so the if branch should be outside the `step_copy` function. The `step` functions returned in the builder are what are actually run when you do `sim.run`, so they should be written to be as fast as possible.

Additionally, it should only be necessary to use `at` when `dst_slice` or `src_slice` are lists (or numpy arrays) of integers, correct? You could check for that also before defining the step function so that we only use `at` when we need to.

Reply:
If we really want to be specific, the only time we need ``at`` is when ``dst_slice`` is a list or ndarray with repeated indices. We could check for this with something like ``len(np.unique(dst_slice)) < len(dst_slice)``.